### PR TITLE
feat(ais-relevant-sort): add widget

### PR DESCRIPTION
**Pull request by eunjae-lee**
## Summary

This PR adds `ais-smart-sort` widget.

**Related task:**
- [x] `stats` widget

**In other flavors:**
* https://github.com/algolia/instantsearch.js/pull/4648
* https://github.com/algolia/react-instantsearch/pull/3009

### CODE REVIEW

File: stories/SmartSort.stories.js
Comment (eunjae-lee) [inline stories/SmartSort.stories.js:21]: Not sure what I did wrong. This doesn't show up.

File: package.json
Comment (eunjae-lee) [inline package.json:49]: This is temporary

**Review by Haroenv — APPROVED** [review]

**Review by shortcuts — APPROVED** [review]
LGTM! We should probably wait until the final name has been decided to merge it, wdyt?

**Pull request merged**
